packages/venia-ui/lib/components/MegaMenu/megaMenuItem.js

```diff
@@ -14,26 +19,76 @@ import PropTypes from 'prop-types';
  * @param {int} props.mainNavWidth - width of the main nav. It's used for setting min-width of the submenu
  */
 const MegaMenuItem = props => {
-    const { activeCategoryId, category, mainNavWidth } = props;
+    const {
+        activeCategoryId,
+        category,
+        mainNavWidth,
+        subMenuState,
+        disableFocus
+    } = props;
     const classes = useStyle(defaultClasses, props.classes);
     const categoryUrl = resourceUrl(
         `/${category.url_path}${category.url_suffix || ''}`
     );
 
-    const children = category.children.length ? (
-        <Submenu items={category.children} mainNavWidth={mainNavWidth} />
-    ) : null;
-    const isActive = category.id === activeCategoryId;
+    const talonProps = useMegaMenuItem({
+        category,
+        activeCategoryId,
+        subMenuState,
+        disableFocus
+    });
+
+    const {
+        isFocused,
+        isActive,
+        handleCloseSubMenu,
+        isMenuActive,
+        a11yClick,
+        toggleSubMenu
+    } = talonProps;
+
+    const megaMenuItemClassname = isMenuActive
+        ? classes.megaMenuItem_active
+        : classes.megaMenuItem;
+
+    const children = useMemo(() => {
+        return category.children.length ? (
+            <Submenu
+                isFocused={isFocused}
+                subMenuState={subMenuState}
+                items={category.children}
+                mainNavWidth={mainNavWidth}
+                handleCloseSubMenu={handleCloseSubMenu}
+            />
+        ) : null;
+    }, [category, isFocused, mainNavWidth, subMenuState, handleCloseSubMenu]);
 
     return (
-        <div className={classes.megaMenuItem}>
+        <div className={megaMenuItemClassname}>
             <Link
+                onKeyDown={e => {
+                    a11yClick(e) && toggleSubMenu(e);
+                }}
                 className={
                     isActive ? classes.megaMenuLinkActive : classes.megaMenuLink
                 }
                 to={categoryUrl}
             >
                 {category.name}
+                {category.children.length ? (
+                    <Icon
+                        className={classes.arrowDown}
+                        src={ArrowDown}
+                        size={17}
```

Comment:
Any reason for the extra pixel? I would expect this to match the font-size of the menu item, which is defaulting to 1rem (16px).